src/target/riscv/riscv-013.c

```diff
@@ -1145,15 +1171,28 @@ static int register_write_abstract(struct target *target, enum gdb_regno number,
 			!info->abstract_write_csr_supported)
 		return ERROR_FAIL;
 
-	uint32_t command = access_register_command(target, number, size,
+	const unsigned int size_bits = register_size(target, number);
+	const uint32_t command = access_register_command(target, number, size_bits,
 			AC_ACCESS_REGISTER_TRANSFER |
 			AC_ACCESS_REGISTER_WRITE);
+	LOG_DEBUG_REG(target, AC_ACCESS_REGISTER, command);
+	assert(size_bits % 32 == 0);
+	const unsigned int size_in_words = size_bits / 32;
+	/** The batch consists of:
+	 * * A write of "data*" ("size_in_words" scans).
+	 * * A write of "command" (1 scan).
+	 * * A read of "abstractcs" (1 scan).
+	 * * An implicit "nop".
+	 */
```

Comment:
Nitpick: 
- The comment does not need to start with the doxygen mark `/**`.
- Using `-` as bullet points makes it visually more clear than the stars that are already used at the beginning of each comment line.

```suggestion
	/* The batch consists of:
	 * - A write of "data*" ("size_in_words" scans).
	 * - A write of "command" (1 scan).
	 * - A read of "abstractcs" (1 scan).
	 * - An implicit "nop".
	 */
```

Reply:
Addressed.